radicle-node/src/worker/fetch.rs

```diff
@@ -0,0 +1,322 @@
+    /// references.
+    pub fn refspecs(&self) -> Vec<Refspec<git::PatternString, git::PatternString>> {
+        let id = git::PatternString::from(IDENTITY_BRANCH.clone().into_refstring());
+        match self.repo {
+            StagedRepository::Cloning(_) => Refspec {
+                src: id.clone(),
+                dst: id,
+                force: false,
+            }
+            .into_refspecs(),
+            StagedRepository::Fetching(_) => Rad(self.namespaces.clone()).into_refspecs(),
+        }
+    }
+
+    /// Convert the [`StagingPhaseInitial`] into [`StagingPhaseFinal`] to continue
+    /// the fetch process.
+    pub fn into_final(self) -> Result<StagingPhaseFinal<'a>, error::Transition> {
+        let remotes = match &self.repo {
+            StagedRepository::Cloning(repo) => {
+                log::debug!(target: "worker", "Loading remotes for clone");
+                let oid = ReadRepository::identity_head(repo)?;
+                log::trace!(target: "worker", "Loading 'rad/id' @ {oid}");
+                let (doc, _) = Doc::<Unverified>::load_at(oid, repo)?;
+                let doc = doc.verified()?;
+                doc.delegates.map(PublicKey::from)
+            }
+            StagedRepository::Fetching(repo) => {
+                log::debug!(target: "worker", "Loading remotes for fetching");
+                match self.namespaces.clone() {
+                    // Nb. Namespaces::One is not constructed in
+                    // namespaces_for so it's safe to just bundle this
+                    // with Namespaces::All
+                    Namespaces::One(_) | Namespaces::All => {
+                        let mut remotes = repo.delegates()?.map(PublicKey::from);
+                        remotes.extend(repo.remote_ids()?.collect::<Result<Vec<_>, _>>()?);
+                        remotes
+                    }
+                    Namespaces::Many(remotes) => remotes,
+                }
+            }
+        };
+
+        Ok(StagingPhaseFinal {
+            repo: self.repo,
+            production: self.production,
+            remotes,
+            _tmp: self._tmp,
+        })
+    }
+
+    fn repository(
+        staging: &Storage,
+        production: &Storage,
+        rid: Id,
+    ) -> Result<StagedRepository, error::Setup> {
+        match production.contains(&rid) {
+            Ok(true) => {
+                let url = url::File::new(production.path_of(&rid)).to_string();
+                log::debug!(target: "worker", "Setting up fetch for existing repository: {}", url);
+                let to = storage::git::paths::repository(&staging, &rid);
+                let copy = git::raw::build::RepoBuilder::new()
+                    .bare(true)
+                    .clone_local(git::raw::build::CloneLocal::Local)
+                    .clone(&url, &to)?;
+                // Nb.: need to clean up origin created during clone
+                copy.remote_delete("origin")?;
```

Comment:
Any harm in leaving it? It is annoying that it gets created :thinking: -- but if it has no consequences, I wonder if we can leave it because it seems random to delete a remote here that we didn't manually create.

Reply:
It was initially causing me issues -- I can't remember why but it was something to do with `HEAD` or `FETCH_HEAD`. Everything works removing it though